themes/startwords/assets/js/notes.js

```diff
@@ -1,170 +1,243 @@
     }
-    window.setTimeout(function(){
-        setupNote();
+
+    disableHideOnScroll(seconds) {
+        // temporarily disable note hide on scroll
+        ContextualNotes.hideOnScroll = false;
+        window.setTimeout(function(){ ContextualNotes.hideOnScroll = true;}, seconds*1000);
+    }
+
+    get currentNote() {
+        // find currently selected note, if there is one
+        if (location.hash.startsWith('#fn:')) {
+            return document.getElementById(location.hash.slice(1));
+        }
+    }
+
+    showNote() {
+        // display the current note (if one is targeted), and bind backref handler
+        var note = this.currentNote;
         if (note) {
-            note.opacity = 1;
+            // if a note occurs near the end of the document,
+            // end notes may already be visible.
+            if (this.endnotes.classList.contains('endnotes')) {
+                // hide end notes so the note can be shown as a contextual note
+                this.endnotes.classList.remove('endnotes');
+                // set a flag that endnotes need to be re-shown
+                ContextualNotes.showEndnotesOnClose = true;
+                // temporarily disable hide on scroll, since
+                // hiding the endnotes changes document length
+                // and can trigger scrolling
+                this.disableHideOnScroll(2);
+            }
+
+            this.positionContextualNote(note);
+            // get back reference link to override behavior
+            var backref = note.getElementsByClassName('footnote-backref').item(0);
+            backref.addEventListener('click', ContextualNotes.noteCloseLinkHandler);
         }
-    }, 500);
+    }
 
-    // create observable to set endnotes class on footnotes ol when visible
-    var observer = new IntersectionObserver(endnoteIntersectionCallback);
-    var endnotes = document.querySelector('.footnotes');
-    if (endnotes) observer.observe(endnotes);
-});
+    static closeNote(note) {
+        note = (typeof note !== 'undefined') ?  note : document.getElementById(location.hash.slice(1));
+        if (note) {
+            // change the location hash, deselecting the note
+            // Set to non-existent id so note is no longer targeted,
+            // but document does not scroll
+            location.hash = '#-';
+            // unbind the close link handler
+            var backref = note.getElementsByClassName('footnote-backref').item(0);
+            backref.removeEventListener('click', ContextualNotes.noteCloseLinkHandler);
+
+            // if endnotes were hidden to show this note, redisplay them
+            if (ContextualNotes.showEndnotesOnClose) {
+                ContextualNotes.showEndnotes()
+                ContextualNotes.showEndnotesOnClose = false;
+            }
+        }
+    }
 
-window.addEventListener('hashchange', function() {
-    setupNote();
-}, false);
+    static noteCloseLinkHandler() {
```

Comment:
think the JS convention is to name handlers things like `onNoteClose`, fwiw

Reply:
Ah, good to know. This is the handler for the x link which is one way to close the note and calls the `closeNote` method. I feel like this name might be confusing — thoughts?

Reply:
yeah, idk what name works best...`click` doesn't cover touch devices, and `link` doesn't totally explain that it's the "X" button (which really is more of a button here). might just need to add docblocks for each method that explain when they are called.

fwiw handler-named functions like `onNoteClose` do usually imply that they happen _prior_ to others, so I would expect something named `onNoteClose()` to be called before the actual call to `closeNote()` (which is indeed what happens).